gpt_bpe.go

```diff
@@ -364,23 +338,27 @@ func NewEncoder(vocabId string) (*GPTEncoder, error) {
 	}
 
 	if specialConfig.EncloseEosBos {
-		tokenizerSpecialConfig.AddBosToken = true
-		tokenizerSpecialConfig.AddEosToken = true
+		eosBosBool := true
+		hfConfig.AddBosToken = &eosBosBool
+		hfConfig.AddEosToken = &eosBosBool
```

Comment:
You really should not be assigning a pointer value to two different variables.

What you are doing here is pointing to the same variable. If someone changes the value, it's going to change for both values.

Create a new `bosBool` value, and rename the prior one to `eosBool`.